swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt

```diff
@@ -490,6 +504,29 @@ abstract class AbstractResourceTypeHandler<T : Resource>(
     Action.DELETE.printResult(workConfiguration, processed = candidateCounter.get(), excluded = excludedCount, total = candidates.size)
   }
 
+  /**
+   * Downstream services may be overloaded or rate limited from other services due to too many task submissions from Swabbie.
+   *
+   * If submitted in-progress tasks are above the configured threshold, wait until the task repository cache is refreshed.
+   */
+  private fun waitForTaskProcessing(taskIds: MutableList<String>, workConfiguration: WorkConfiguration) {
+    if (taskTrackingRepository.getInProgress().count { taskIds.contains(it) } >= workConfiguration.maxTasksPerBatch) {
+      latch = CountDownLatch(1)
+      latch.await()
```

Comment:
Need to test this, not totally sure if this will work -- may just need to do a `sleep` here.  Either way, a bit gross.

Reply:
Wondering if this can be made a little simpler by setting `maxItemsToProcess` to be a difference between the `taskTrackingRepository.getInProgress().count` and `maxTasksPerBatch` , we may not have to sleep or wait then ? thoughts ?

Reply:
Problem with that is we would then only process the `maxTasksPerBatch` per cycle (I should really rename that to `maxTasksInProgressPerBatch`).  My goal is to still process the `maxItemsProcessedPerCycle` but just limit the total number of in-progress tasks (per resource type) on the Orca side.  I'm open to a simpler way though.